src/lib.rs

```diff
@@ -71,7 +71,8 @@ where
 {
     fn message_id(&self) -> u32;
     fn message_name(&self) -> &'static str;
-    fn ser(&self) -> Vec<u8>;
+    fn ser_v1(&self) -> Vec<u8>;
```

Comment:
Why not creating a `ser` function that receives as argument `MavlinkVersion` ? Like what we have for `deser`. The only difference of `ser_v1` and `ser_v2` is the trailing removal logic.